### PR TITLE
feat: Support resource limits in nodePoolOptions

### PR DESCRIPTION
**Pull request by rushtehrani**
**What this PR does**:
Extend options field in nodePool to take resource quotas.

**Which issue(s) this PR fixes**:

Fixes onepanelio/core#392

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/build.go [hunk 1]
```diff
@@ -437,43 +437,28 @@ func runKustomizeBuild(path string) (rm resmap.ResMap, err error) {
...
-				}
-				optionChunk = append(optionChunk, "")
-			}
-			break
-		}
+func generateApplicationNodePoolOptions(nodePoolData *yaml2.Node) string {
+	nodePool := struct {
+		Options []map[string]interface{}
+	}{}
+	err := nodePoolData.Decode(&nodePool)
```
Comment (Vafilor) [inline cmd/build.go:444]: Minor nitpick, but why not do 
``` if err := nodePoolData.Decode(&nodePool); err != nil {
		log.Println(err)
		return ""
}

Comment (rushtehrani) [reply cmd/build.go:444]: Good catch, will fix shortly.

Comment (Vafilor) [reply cmd/build.go:444]: Only thing thing I would say is - we might want to add unit tests for things like this. There looks like a lot of logic in the old one, but I think you've got it covered with your decoding into a struct.

Comment (rushtehrani) [reply cmd/build.go:444]: Agreed

Comment (rushtehrani) [reply cmd/build.go]: Added test

**Pull request merged**